File: God Zuo/class046 构建前缀树的技巧/My06.py
class Solution:
    def minSubarray(self, nums, p):
        mod,sum=0
        for num in nums:
            sum+=num
        mod=sum%p
        if mod==0:
            return 0
        ans=float('inf')
        cur=0
        book={0:-1}
        for i in range(nums):
            cur=(cur+nums[i])%p
            find=(cur-mod+p)%p
            if find in book:
                ans=min(ans,i-book[find])
            book[cur]=i
        return -1 if ans==len(nums) else ans
    

File: God Zuo/class071 子数组的最大累加和下/My06.cpp
// 删掉1个数字后长度为k的子数组最大累加和
// 给定一个数组nums，求必须删除一个数字后的新数组中
// 长度为k的子数组最大累加和，删除哪个数字随意
// 对数器验证

#include<bits/stdc++.h>
using namespace std;

int wd[10010];

int f2(vector<int>& nums,int k){
    int n = nums.size();
    if(n<=k){
        return 0;
    }
    int l = 0, r = 0;
    long long sum = 0;
    int ans = INT_MIN;
    for (int i = 0; i < n;++i){
        while(l<r && nums[wd[r-1]]>=nums[i])
            --r;
        wd[r++] = i;
        sum += nums[i];
    }
}